### PR TITLE
chore: temporary allow http always

### PR DESCRIPTION
**Pull request by tustvold**
The auth system uses the same reqwest client, which is causing the metadata requests to fail with `builder error for url (http://169.254.169.254/latest/api/token): URL scheme is not allowed`

### CODE REVIEW

**Review by crepererum — APPROVED** [review]

File: clap_blocks/src/object_store.rs [hunk 1]
```diff
@@ -280,7 +280,8 @@ fn new_gcs(_: &ObjectStoreConfig) -> Result<Arc<DynObjectStore>, ParseError> {
 #[cfg(feature = "aws")]
 fn new_s3(config: &ObjectStoreConfig) -> Result<Arc<DynObjectStore>, ParseError> {
     let mut builder = AmazonS3Builder::new()
-        .with_allow_http(config.aws_allow_http)
+        // TODO: Temporary fix
```
Comment (alamb) [inline clap_blocks/src/object_store.rs:283]: This needs a ticket tracking what is wrong

**Review by lukebond — APPROVED** [review]
i don't have full context but guessing reqwest does auth with http and the new builder doesn't have it on by default. seems reasonable as a quick fix for prod

**Comment by alamb** [comment]
Why are we doing authentication over `http` -- I am sure I am missing something but that seems like a bad idea

**Comment by tustvold** [comment]
> Why are we doing authentication over http -- I am sure I am missing something but that seems like a bad idea

The instance metadata endpoint is a loopback IP that is over http

File: clap_blocks/src/object_store.rs (hunk 1 above)
Comment (tustvold) [reply clap_blocks/src/object_store.rs:283]: I'll just fix it upstream, this was just a quick get prod unblocked

Comment (tustvold) [reply clap_blocks/src/object_store.rs:283]: https://github.com/influxdata/influxdb_iox/pull/5359

**Pull request merged**